src/java.base/unix/native/libnio/fs/UnixNativeDispatcher.c

```diff
@@ -117,7 +208,7 @@ static jfieldID attrs_st_mtime_nsec;
 static jfieldID attrs_st_ctime_sec;
 static jfieldID attrs_st_ctime_nsec;
 
-#ifdef _DARWIN_FEATURE_64_BIT_INODE
+#if defined(_DARWIN_FEATURE_64_BIT_INODE) || defined(__linux__)
 static jfieldID attrs_st_birthtime_sec;
```

Comment:
birthtime_nsec missing? Just set it to zero on darwin and use the value under linux?

Reply:
I've deliberately omitted that change here, not sure if this change of behaviour is wanted as it would add a field in `UnixFileAttributes` (it was part of the v2 patch). @bplb @AlanBateman Thoughts? I can include it of course, but don't want to go in circles.

Reply:
Understood, seems just a little bit silly that a limitation in a different kernel (darwin) artificially limits the birthtime here, while it easily accessible with statx. Otherwise this patch series looks quite good!

Reply:
Well, I've reconsidered. It's fairly trivial to support, so I've added it. The latest patch does this. It's an incremental webrev change so we can go back if not desired.